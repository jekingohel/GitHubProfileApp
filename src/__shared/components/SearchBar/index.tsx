import {PropsWithChildren} from 'react';
import {TextInput} from 'react-native';
import {Text, View} from '..';
import { TouchableOpacity } from 'react-native';

type SearchPropsType = PropsWithChildren<{
  handleOnChange?: (arg: any) => any;
  inputRef?: any;
  onSubmit?: any
}>;

function SearchBar({
  handleOnChange,
  inputRef,
  onSubmit
}: SearchPropsType): React.JSX.Element {
  return (
    <View
      transparent
      flexDirection="row"
      alignItems="center"
      justifyContent="space-between"
      style={{
        padding: 10,
        columnGap: 15,
        borderBottomWidth: 1,
        borderBottomColor: '#F4F4F4',
      }}>
      <TextInput
        ref={inputRef}
        editable
        autoCapitalize="none"
        clearButtonMode="always"
        style={{
          backgroundColor: '#f4f4f4',
          fontSize: 15,
          borderWidth: 0,
          flexShrink: 1,
          flexGrow: 1,
          padding: 10,
          borderRadius: 10,
        }}
        onChangeText={handleOnChange}
        placeholder="Search User"
        testID="search-input"
      />
      <TouchableOpacity activeOpacity={0.5} onPress={onSubmit} testID="submit-button">
        <Text fontSize={18} lineHeight={30} color="#3399ff">
          Submit
        </Text>
      </TouchableOpacity>
    </View>
  );
}

export default SearchBar;
