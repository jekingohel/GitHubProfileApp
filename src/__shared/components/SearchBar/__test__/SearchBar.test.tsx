import React from 'react';
import renderer from 'react-test-renderer';
import {render} from '@testing-library/react-native';
import SearchBar from '..';

test('renders search bar components', () => {
  // Render the SearchBar component

  const {getByPlaceholderText, getByTestId, toJSON} = render(
    <SearchBar  />,
  );

  // Assert the presence of the search input
  const searchInput = getByPlaceholderText('Search User');
  expect(searchInput).toBeTruthy();

  // Assert the presence of the submit button
  const submitButton = getByTestId('submit-button');
  expect(submitButton).toBeTruthy();
  expect(toJSON()).toMatchSnapshot();
});


