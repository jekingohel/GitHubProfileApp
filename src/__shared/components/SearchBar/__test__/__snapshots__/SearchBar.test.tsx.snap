// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders search bar components 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "transparent",
      "borderBottomColor": "#F4F4F4",
      "borderBottomWidth": 1,
      "columnGap": 15,
      "flexDirection": "row",
      "justifyContent": "space-between",
      "padding": 10,
    }
  }
  testID="test-view"
>
  <TextInput
    autoCapitalize="none"
    clearButtonMode="always"
    editable={true}
    placeholder="Search User"
    style={
      {
        "backgroundColor": "#f4f4f4",
        "borderRadius": 10,
        "borderWidth": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "fontSize": 15,
        "padding": 10,
      }
    }
    testID="search-input"
  />
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
    testID="submit-button"
  >
    <Text
      style={
        {
          "color": "#3399ff",
          "fontSize": 18,
          "fontStyle": "normal",
          "fontWeight": "400",
          "lineHeight": 30,
          "textAlign": "left",
        }
      }
    >
      Submit
    </Text>
  </View>
</View>
`;
